### PR TITLE
es6

### PR DESCRIPTION
**Pull request by ashleyschuett**
@nicktate @normanjoyner

### CODE REVIEW

**Comment by ashleyschuett** [comment]
am I allowed to change the image in the .drone.yml?

**Comment by nicktate** [comment]
You can change it but you have to resecure it. If you change it ill update your PR with the new secured file

**Comment by nicktate** [comment]
this PR is horrifying FYI

**Comment by ashleyschuett** [comment]
:(

File: handlers/v1/application.js [hunk 1]
```diff
@@ -1,62 +1,64 @@
...
-                        "env_vars",
-                        "image",
-                        "memory",
-                        "network_mode",
-                        "respawn",
-                        "privileged",
-                        "tags",
-                        "volumes"
+        create(req, res, next) {
+            return core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join('::'), (err/*, application*/) => {
```
Comment (nicktate) [inline handlers/v1/application.js]: not a problem, but note to self, we should add in a function `exists` to myriad @normanjoyner

File: handlers/v1/application.js [hunk 2]
```diff
@@ -141,163 +154,164 @@ module.exports = function(core){
...
-                        });
-                    }
-                    catch(err){
-                        res.stash.code = 400;
-                    }
+                    return next();
                 }
 
-                return next();
+                try{
```
Comment (nicktate) [inline handlers/v1/application.js]: space after try

File: handlers/v1/application.js [hunk 3]
```diff
@@ -141,163 +154,164 @@ module.exports = function(core){
...
-                                res.stash.code = 200;
-                            }
-                        });
-                    }
-                    catch(err){
-                        res.stash.code = 400;
-                    }
+                    return next();
+                }
+                try{
```
Comment (nicktate) [inline handlers/v1/application.js]: newline before try and space after

File: handlers/v1/applications.js [hunk 4]
```diff
@@ -1,35 +1,40 @@
...
+                if(err) {
+                    return next();
+                }
 
+                return async.each(applications, (application_name, fn) => {
+                    return core.cluster.myriad.persistence.get(application_name, (err, application) => {
+                        if(err) {
+                            return fn();
+                        }
                         try{
```
Comment (nicktate) [inline handlers/v1/applications.js]: newline before try and space after

File: handlers/v1/applications.js [hunk 5]
```diff
@@ -1,35 +1,40 @@
...
                             application = JSON.parse(application);
-                            core.applications.get_containers(application.id, function(err, containers){
-                                if(err)
+                            return core.applications.get_containers(application.id, (err, containers) => {
+                                if(err) {
                                     return fn();
-
+                                }
```
Comment (nicktate) [inline handlers/v1/applications.js]: newline after

File: handlers/v1/applications.js [hunk 6]
```diff
@@ -38,47 +43,51 @@ module.exports = function(core){
         },
 
         // create applications
-        create: function(req, res, next){
-            if(!_.isUndefined(req.body)){
-                var all_applications;
+        create(req, res, next) {
+            if(!_.isUndefined(req.body)) {
+                let all_applications;
```
Comment (nicktate) [inline handlers/v1/applications.js]: `let all_applications = null;`

**Comment by nicktate** [comment]
@ashleyschuett Regarding the core, maybe you could add a global preprocess middleware that attaches core to the req object? Then just do `req.core` in your handlers to make that a tad cleaner?

**Comment by nicktate** [comment]
@ashleyschuett If you do something like that cleaning up how core is pass around please make it a separate branch & PR.

File: handlers/v1/host.js [hunk 7]
```diff
@@ -1,44 +1,45 @@
...
+            }
             host.containers = [];
 
-            core.cluster.myriad.persistence.keys([core.constants.myriad.CONTAINERS_PREFIX, "*", "*"].join("::"), function(err, containers){
-                if(err){
+            return core.cluster.myriad.persistence.keys([core.constants.myriad.CONTAINERS_PREFIX, '*', '*'].join('::'), (err, containers) => {
+                if(err) {
                     res.stash.code = 400;
-                    return fn();
+                    return next();
```
Comment (nicktate) [inline handlers/v1/host.js]: good catch!

File: handlers/v1/host.js [hunk 8]
```diff
@@ -1,44 +1,45 @@
...
-                async.each(containers, function(container_name, fn){
-                    core.cluster.myriad.persistence.get(container_name, function(err, container){
-                        if(err)
+                return async.each(containers, (container_name, fn) => {
+                    return core.cluster.myriad.persistence.get(container_name, (err, container) => {
+                        if(err) {
                             return fn();
-
+                        }
                         try{
```
Comment (nicktate) [inline handlers/v1/host.js]: newline before try and space after

Comment (nicktate) [inline handlers/v1/host.js]: just realized this probably invalidates the eslint, can you update the rule to allow a space after try if so

File: handlers/v1/host.js [hunk 9]
```diff
@@ -47,40 +48,40 @@ module.exports = function(core){
...
+        update(req, res, next) {
+            if(!_.isUndefined(req.body) && _.has(req.body, 'tags')) {
+                let hosts = _.indexBy(core.cluster.legiond.get_peers(), 'id');
 
-                var attributes = core.cluster.legiond.get_attributes();
+                let attributes = core.cluster.legiond.get_attributes();
                 hosts[attributes.id] = attributes;
 
-                var host = hosts[req.params.host];
+                let host = hosts[req.params.host];
```
Comment (nicktate) [inline handlers/v1/host.js]: most of these lets could be `const`

**Comment by nicktate** [comment]
@ashleyschuett Also in a separate PR update lodash to 4.x

File: handlers/v1/hosts.js [hunk 10]
```diff
@@ -1,59 +1,67 @@
...
+
+module.exports = function(core) {
 
     return {
         // get all hosts
-        get: function(req, res, next){
-            var hosts = _.indexBy(core.cluster.legiond.get_peers(), "id");
-            var attributes = core.cluster.legiond.get_attributes();
+        get(req, res, next) {
+            let hosts = _.indexBy(core.cluster.legiond.get_peers(), 'id');
```
Comment (nicktate) [inline handlers/v1/hosts.js]: revisit your lets across the board, many of them could be `const`. I won't comment on them anymore

File: handlers/v1/hosts.js [hunk 11]
```diff
@@ -1,59 +1,67 @@
...
+
+                return async.each(applications, (application_name, fn) => {
+                    // eslint thinks configuration isn't used since it's in a try/catch
+                    // eslint-disable-next-line no-unused-vars
+                    return core.cluster.myriad.persistence.get(application_name, (err, configuration) => {
+                        if(err) {
                             return fn();
+                        }
 
                         try{
```
Comment (nicktate) [inline handlers/v1/hosts.js]: space after try

**Comment by ashleyschuett** [comment]
@nicktate the PR is updated. I'll make other PRs for the other things requested. Also drone is updated, just need you to make/update the .drone.sec file.

File: handlers/v1/application.js [hunk 12]
```diff
@@ -1,62 +1,65 @@
...
+module.exports = function(core) {
 
     return {
         // get application
-        get: function(req, res, next){
-            core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join("::"), function(err, application){
-                if(err && err.name == core.constants.myriad.ENOKEY)
+        get(req, res, next) {
+            return core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join('::'), (err, application) => {
+                if(err && err.name == core.constants.myriad.ENOKEY) {
```
Comment (normanjoyner) [inline handlers/v1/application.js]: `===`

File: handlers/v1/application.js [hunk 13]
```diff
@@ -1,62 +1,65 @@
...
+
+module.exports = function(core) {
 
     return {
         // get application
-        get: function(req, res, next){
-            core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join("::"), function(err, application){
-                if(err && err.name == core.constants.myriad.ENOKEY)
+        get(req, res, next) {
+            return core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join('::'), (err, application) => {
```
Comment (normanjoyner) [inline handlers/v1/application.js]: use myriad delimiter constant

File: handlers/v1/application.js (hunk 1 above)
Comment (normanjoyner) [inline handlers/v1/application.js]: :+1: tickets in for this

File: handlers/v1/application.js [hunk 14]
```diff
@@ -1,62 +1,65 @@
...
-                        "memory",
-                        "network_mode",
-                        "respawn",
-                        "privileged",
-                        "tags",
-                        "volumes"
+        create(req, res, next) {
+            return core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join('::'), (err/*, application*/) => {
+                if(err && err.name == core.constants.myriad.ENOKEY) {
+                    const config = _.pick(req.body, [
```
Comment (normanjoyner) [inline handlers/v1/application.js]: alphabetize this list

File: handlers/v1/application.js [hunk 15]
```diff
@@ -66,66 +69,77 @@ module.exports = function(core){
...
-                if(err){
+        update(req, res, next) {
+            return core.cluster.myriad.persistence.keys(core.constants.myriad.APPLICATIONS, (err, applications) => {
+                if(err) {
                     res.stash.code = 400;
                     return next();
                 }
 
-                if(!_.contains(applications, [core.constants.myriad.APPLICATION_PREFIX, req.params.application].join("::"))){
+                if(!_.contains(applications, [core.constants.myriad.APPLICATION_PREFIX, req.params.application].join('::'))) {
```
Comment (normanjoyner) [inline handlers/v1/application.js]: use myriad delimiter constant

File: handlers/v1/application.js [hunk 16]
```diff
@@ -141,163 +155,165 @@ module.exports = function(core){
...
                 return next();
             });
         },
 
         // get application containers
-        get_containers: function(req, res, next){
-            core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join("::"), function(err, application){
-                if(err && err.name == core.constants.myriad.ENOKEY)
+        get_containers(req, res, next) {
+            return core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join('::'), (err, application) => {
```
Comment (normanjoyner) [inline handlers/v1/application.js]: use myriad delimiter constant

File: handlers/v1/application.js [hunk 17]
```diff
@@ -141,163 +155,165 @@ module.exports = function(core){
...
+                }
             });
         },
 
         // get application container
-        get_container: function(req, res, next){
-            core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join("::"), function(err, application){
-                if(err && err.name == core.constants.myriad.ENOKEY)
+        get_container(req, res, next) {
+            return core.cluster.myriad.persistence.get([core.constants.myriad.APPLICATION_PREFIX, req.params.application].join('::'), (err, application) => {
```
Comment (normanjoyner) [inline handlers/v1/application.js]: use myriad delimiter constant

File: handlers/v1/applications.js [hunk 18]
```diff
@@ -38,47 +45,51 @@ module.exports = function(core){
...
+
+                return async.series([
+                    (fn) => {
+                        return core.cluster.myriad.persistence.keys(core.constants.myriad.APPLICATIONS, (err, applications) => {
+                            if(err) {
+                                return fn();
+                            }
+
+                            applications = _.map(applications, (application) => {
+                                return _.last(application.split('::'));
```
Comment (normanjoyner) [inline handlers/v1/applications.js]: use myriad delimiter constant

File: handlers/v1/host.js [hunk 19]
```diff
@@ -1,44 +1,46 @@
...
+            const host = hosts[req.params.host];
+            if(_.isUndefined(host)) {
                 return next();
-
+            }
             host.containers = [];
 
-            core.cluster.myriad.persistence.keys([core.constants.myriad.CONTAINERS_PREFIX, "*", "*"].join("::"), function(err, containers){
-                if(err){
+            return core.cluster.myriad.persistence.keys([core.constants.myriad.CONTAINERS_PREFIX, '*', '*'].join('::'), (err, containers) => {
```
Comment (normanjoyner) [inline handlers/v1/host.js]: use myriad delimiter constant

File: handlers/v1/host.js [hunk 20]
```diff
@@ -1,44 +1,46 @@
...
                             return fn();
+                        }
 
-                        try{
+                        try {
                             container = JSON.parse(container);
-                            if(container.host == host.id){
-                                container.application = container_name.split("::")[2];
+                            if(container.host == host.id) {
+                                container.application = container_name.split('::')[2];
```
Comment (normanjoyner) [inline handlers/v1/host.js]: use myriad delimiter constant

File: handlers/v1/hosts.js [hunk 21]
```diff
@@ -1,59 +1,67 @@
...
+                }, (err) => {
+                    if(err) {
                         res.stash.code = 400;
-                        return fn();
+                        return next();
                     }
 
-                    core.cluster.myriad.persistence.keys([core.constants.myriad.CONTAINERS_PREFIX, "*", "*"].join("::"), function(err, containers){
-                        if(err){
+                    return core.cluster.myriad.persistence.keys([core.constants.myriad.CONTAINERS_PREFIX, '*', '*'].join('::'), (err, containers) => {
```
Comment (normanjoyner) [inline handlers/v1/hosts.js]: use myriad delimiter constant

File: handlers/v1/hosts.js [hunk 22]
```diff
@@ -1,59 +1,67 @@
...
+                            return core.cluster.myriad.persistence.get(container_name, (err, container) => {
+                                if(err) {
                                     return fn();
-
-                                try{
+                                }
+                                try {
                                     container = JSON.parse(container);
-                                    var application = container_name.split("::")[2];
+                                    const application = container_name.split('::')[2];
```
Comment (normanjoyner) [inline handlers/v1/hosts.js]: use myriad delimiter constant

File: .eslintignore [hunk 23]
```diff
@@ -0,0 +1,3 @@
+test/*
+apiDocs/*
```
Comment (normanjoyner) [inline .eslintignore]: remove

File: .eslintignore [hunk 24]
```diff
@@ -0,0 +1,3 @@
...
+apidoc.json
```
Comment (normanjoyner) [inline .eslintignore]: remove

**Comment by ashleyschuett** [comment]
@normanjoyner updated

File: .drone.yml [hunk 25]
```diff
@@ -1,7 +1,8 @@
 build:
-  image: library/node:0.10.41
+  image: library/node:6.3.0
```
Comment (normanjoyner) [inline .drone.yml]: remove and i will make this update separately

File: .drone.yml [hunk 26]
```diff
@@ -1,7 +1,8 @@
...
   commands:
     - npm install
+    - npm run lint
```
Comment (normanjoyner) [inline .drone.yml]: remove and i will make this update separately

**Pull request merged**